src/test/java/org/apache/bcel/util/ClassPathRepositoryTestCase.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.apache.bcel.util;
+
+import java.io.IOException;
+
+import org.apache.bcel.classfile.JavaClass;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests {@link ClassPathRepository} and {@link MemorySensitiveClassPathRepository}.
+ *
+ * <p>Without memory scarcity, these two classes behave in the same manner.
+ */
+public class ClassPathRepositoryTestCase {
+
+    private void verifyCaching(AbstractClassPathRepository repository) throws ClassNotFoundException {
```

Comment:
Please run `mvn -P jacoco -P japicmp install site' and make sure you get as close to 100% code coverage in the reports for the new and updated code. Also make sure the JApiCmp report does not report BC breakage.

Reply:
`mvn site` is failing. Created #30 .

Reply:
Added more tests to increase test coverage. Added coverage information in the description of this PR.

I found corner cases with IOException were hard to cover.

![image](https://user-images.githubusercontent.com/28604/60915082-f7386200-a258-11e9-929b-a73b2a440bd8.png)